### ISSUE TITLE
Redirect logic is broken after upgrading to v9.1.0

### ISSUE THREAD

**Issue by hnvn** [Milad-Akarie/auto_route_library#2020]
I currently use this function `RoutingController.buildPageRoute(path)` to parse deeplink and handle routing logic manually.

In v8.3.0, this function works well with redirect path but after I upgraded to v9.1.0, the function no longer work with redirect path anymore. Do I need to update anything in my codes to adapt the changing in function `buildPageRoute()` for redirection logic?

**Comment by Milad-Akarie** [comment]
@hnvn I don't see how the latest refactors would effect this method, make sure you re-generated the routes after upgrading, if that doesn't wok please provide an example to repreduce this

**Comment by a-h-mzd** [comment]
It's not just that. The `RedirectRoute` is also broken in the v9 release.

### Sample router code:
```dart
@AutoRouterConfig()
final class AppRouter extends RootStackRouter {
  @override
  List<AutoRoute> get routes {
    return <AutoRoute>[
      RedirectRoute(
        path: '/test',
        redirectTo: '/',
      ),
      RedirectRoute(
        path: '/test2',
        redirectTo: '/',
      ),
      AutoRoute(
        path: '/',
        page: HomeRoute.page,
      ),
    ];
  }
}
```

Seems like only the last `RedirectRoute` works and all the previous `RedirectRoute`s are ignored!

**Comment by a-h-mzd** [comment]
For example if you have a wildcard `RedirectRoute` at the end of your routes list, all the url will be redirected even if another route matches that url. (when reloading webpage)

**Comment by Raviteja11122** [comment]
For web also it's broken,

RangeError: Maximum call stack size exceeded
dart:sdk_internal 26143:20                                                  length]
dart:sdk_internal 25809:23                                                  clear]
packages/auto_route/src/router/widgets/deferred_widget.dart.lib.js 5644:28  [_match]
packages/auto_route/src/router/widgets/deferred_widget.dart.lib.js 5678:26  [_handleRedirect]

**Comment by xleon** [comment]
Similar issue here. Redirect not working at all for a deeplink path

![image](https://github.com/user-attachments/assets/2c6e719b-e488-4c9d-8411-5176cf6113a4)

When deeplink happens, the router goes straight to `/`

I have a wildcard at the end: 
`RedirectRoute(path: '*', redirectTo: '/'),`

**Comment by Gyula-UpScale** [comment]
I'd also like to add that this configuration does not load the "/courses" path when the "/" path is requested on Flutter Web: (I'm on auto_route 9.2.0, auto_route_generator: 9.0.0)
`AutoRoute(path: '/', page: DashboardRoute.page, children: [
          RedirectRoute(path: '', redirectTo: 'courses'), 
          CustomRoute(
              path: 'courses',
              initial: true,
              page: CoursesRoute.page,
              transitionsBuilder: null),
...
`

**Comment by SamerOrfali22** [comment]
samer here

**Comment by hnvn** [comment]
confirmed that it is fixed in v9.2.2
